expertise/execute_expertise.py

```diff
@@ -158,6 +158,56 @@ def execute_expertise(config):
                 scores_path=Path(config['model_params']['scores_path']).joinpath(config['name'] + '_sparse.csv')
             )
 
+    if 'submission_group' in config:
+        # Fetch scores
+        scores = {}
+        original_score_path = Path(config['model_params']['scores_path']).joinpath(config['name'] + '.csv')
+        with open(original_score_path, 'r') as f:
+            csv_reader = csv.reader(f)
+            for line in csv_reader:
+                paper_id = line[0]
+                ac_id = line[1]
+                score = line[2]
+                if paper_id not in scores:
+                    scores[paper_id] = {}
+                scores[paper_id][ac_id] = score
+
+        dataset_root = Path(config.get('dataset', {}).get('directory', './'))
+        # Fetch archive members
+        archive_members = []
+        archive_files = os.listdir(dataset_root.joinpath('archives'))
+        for author_file in archive_files:
+            archive_members.append(author_file.split('.')[0])
+
+        # Fetch submission members
+        with open(dataset_root.joinpath('publications_by_profile_id.json'), 'r') as f:
+            publications_by_profile_id = json.load(f)
+        submission_members = publications_by_profile_id.keys()
+
+        # Perform averaging
+        average_score = {}
```

Comment:
Not really sure, but this averaging part can be made more efficient using the pandas `grouping_by` method. Take a look and see if it helps or not.